pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -100,10 +100,18 @@ func (service *SecurityPolicyService) CreateOrUpdateSecurityPolicy(obj *v1alpha1
 	}
 
 	existingSecurityPolicy := service.securityPolicyStore.GetByKey(*nsxSecurityPolicy.Id)
-	existingRules := service.ruleStore.GetByIndex(common.TagScopeSecurityPolicyCRUID, string(obj.UID))
-	existingGroups := service.groupStore.GetByIndex(common.TagScopeSecurityPolicyCRUID, string(obj.UID))
-
+	tmpExistingRules := service.ruleStore.GetByIndex(common.TagScopeSecurityPolicyCRUID, string(obj.UID))
```

Comment:
We can also change RulesToComparable, ComparableToRule, GroupsToComparable, ComparableToGroups to use pointer as well. So, we  don't need to use temporary rule pointer list here and convert them to teh ruls. We can directly use rule pointers in the compare related to function,  as we did for SecurityPolicyToComparable.

Reply:
Done